src/Indicator.vala

```diff
@@ -27,14 +27,41 @@ public class Power.Indicator : Wingpanel.Indicator {
     private Services.Device primary_battery;
     private bool notify_battery = false;
 
+    private Widgets.ScreenBrightness screen_brightness;
+
     public Indicator (bool is_in_session) {
         Object (code_name : Wingpanel.Indicator.POWER,
                 display_name : _("Power"),
                 description: _("Power indicator"));
 
-        this.is_in_session = is_in_session;
+        display_widget = new Widgets.DisplayWidget ();
+        screen_brightness = new Widgets.ScreenBrightness ();
+        popover_widget = new Widgets.PopoverWidget (screen_brightness, is_in_session);
+
+        popover_widget.settings_shown.connect (() => close ());
+
+        var dm = Services.DeviceManager.get_default ();
```

Comment:
Why has this whole code section been duplicated from `get_widget`?

Reply:
Until I copied it didn't work. And I have doubts, is `get_widget` and `get_display_widget` necessary since I'll creating new instances in `constructor`? Is it safe to delete them, or are they necessary because of Wingpanel API?

Reply:
You can ignore questions, already looked at libwingpanel.